skyportal/models.py

```diff
@@ -240,13 +262,19 @@ class Comment(Base):
     origin = sa.Column(sa.String, nullable=True)
     author = sa.Column(sa.String, nullable=False)
     source_id = sa.Column(sa.ForeignKey('sources.id', ondelete='CASCADE'),
-                          nullable=False, index=True)
-    source = relationship('Source', back_populates='comments')
+                          nullable=True, index=True)
+    source = relationship('Source', foreign_keys=[source_id],
+                          back_populates='comments')
+    candidate_id = sa.Column(
+        sa.ForeignKey("sources.id", ondelete="CASCADE"), nullable=True, index=True
+    )
+    candidate = relationship("Source", foreign_keys=[candidate_id],
```

Comment:
Is it strictly necessary to duplicate the structure here, now that sources and candidates live in the same table?

Reply:
I thought we needed candidate comments to be distinct from source comments